client/app/components/onboardingGuide/GuideStatusPage.js

```diff
@@ -89,7 +90,7 @@ GuideStatusPage.propTypes = {
     }),
     PropTypes.bool.isRequired,
   ]).isRequired,
-  onboarding_data: PropTypes.objectOf(PropTypes.shape({
+  onboarding_data: PropTypes.arrayOf(PropTypes.shape({
     info_image: PropTypes.string,
     cta: PropTypes.string,
     complete: PropTypes.bool.isRequired,
```

Comment:
On another thought, eslint-plugin-react can most likely only complain about missing PropTypes for root-level props, not the data within those.